File: app/src/main/java/ke/choxxy/wordsearch/di/modules/DataSourceModule.java
package ke.choxxy.wordsearch.di.modules;

import android.content.Context;

import ke.choxxy.wordsearch.data.sqlite.DbHelper;
import ke.choxxy.wordsearch.data.sqlite.GameDataSQLiteDataSource;
import ke.choxxy.wordsearch.data.xml.WordXmlDataSource;
import ke.choxxy.wordsearch.data.GameDataSource;
import ke.choxxy.wordsearch.data.WordDataSource;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by abdularis on 18/07/17.
 */

@Module
public class DataSourceModule {

    @Provides
    @Singleton
    DbHelper provideDbHelper(Context context) {
        return new DbHelper(context);
    }

    @Provides
    @Singleton
    GameDataSource provideGameRoundDataSource(DbHelper dbHelper) {
        return new GameDataSQLiteDataSource(dbHelper);
    }

//    @Provides
//    @Singleton
//    WordDataSource provideWordDataSource(DbHelper dbHelper) {
//        return new WordSQLiteDataSource(dbHelper);
//    }

    @Provides
    @Singleton
    WordDataSource provideWordDataSource(Context context) {
        return new WordXmlDataSource(context);
    }

}

File: app/src/main/java/ke/choxxy/wordsearch/di/DataSourceModule.kt
package ke.choxxy.wordsearch.di

import android.content.Context
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import ke.choxxy.wordsearch.data.WordDataSource
import ke.choxxy.wordsearch.data.sqlite.DbHelper
import ke.choxxy.wordsearch.data.sqlite.GameDataSQLiteDataSource
import ke.choxxy.wordsearch.data.xml.WordXmlDataSource
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object DataSourceModule {

    @Provides
    @Singleton
    fun provideGameRoundDataSource(dbHelper: DbHelper): GameDataSQLiteDataSource {
        return GameDataSQLiteDataSource(dbHelper)
    }

    //    @Provides
    //    @Singleton
    //    WordDataSource provideWordDataSource(DbHelper dbHelper) {
    //        return new WordSQLiteDataSource(dbHelper);
    //    }
    @Provides
    @Singleton
    fun provideWordDataSource(@ApplicationContext context: Context): WordDataSource {
        return WordXmlDataSource(context)
    }
}
